app/models/book.rb

```diff
@@ -1,4 +1,15 @@
 class Book < ApplicationRecord
+  has_many :checkouts
+  has_many :users, through: :checkouts
+
   validates :title, presence: true, uniqueness: { case_sensitive: false}
   validates :author, presence: true
+
+  def self.available
+    includes(:checkouts).where(checkouts: { id: nil })
+  end
+
+  def self.unavailable
+    includes(:checkouts).where.not(checkouts: { id: nil })
```

Comment:
@derekprior, It actually turned out to be quite simple. Just so you know, I think it is only supported in Rails 4 and up.